From 62afc10be6851a9bbdf1186ad58679743b9baabf Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:25:14 -0400
Subject: [pci] msi: Return -ENOSYS for unimplemented interfaces, not -1

Message-id: <20140719042514.19438.53670.stgit@gir.stowe>
Patchwork-id: 85986
O-Subject: [RHEL7.1 PATCH 24/81] PCI/MSI: Return -ENOSYS for unimplemented interfaces, not -1
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 8ec5db6b20c860ddd1311c794b38c98ce86ac7ae

commit 8ec5db6b20c860ddd1311c794b38c98ce86ac7ae
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Mon Dec 16 09:34:57 2013 +0100

    PCI/MSI: Return -ENOSYS for unimplemented interfaces, not -1

    Suggested-by: Ben Hutchings <bhutchings@solarflare.com>
    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Reviewed-by: Tejun Heo <tj@kernel.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/pci.h b/include/linux/pci.h
index bb953ea..ba9d304 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -1192,13 +1192,13 @@ struct msix_entry {
 #ifndef CONFIG_PCI_MSI
 static inline int pci_enable_msi_block(struct pci_dev *dev, int nvec)
 {
- return -1;
+ return -ENOSYS;
 }
 
 static inline int
 pci_enable_msi_block_auto(struct pci_dev *dev, int *maxvec)
 {
- return -1;
+ return -ENOSYS;
 }
 
 static inline void pci_msi_shutdown(struct pci_dev *dev)
@@ -1213,7 +1213,7 @@ static inline int pci_msix_table_size(struct pci_dev *dev)
 static inline int pci_enable_msix(struct pci_dev *dev,
       struct msix_entry *entries, int nvec)
 {
- return -1;
+ return -ENOSYS;
 }
 
 static inline void pci_msix_shutdown(struct pci_dev *dev)
-- 
1.7.1